tsdb/wal/watcher.go

```diff
@@ -595,6 +635,23 @@ func (w *Watcher) readCheckpoint(checkpointDir string) error {
 	return nil
 }
 
+func (w *Watcher) CurrentSegment() int {
+	return w.currentSegment
+}
+
+func (w *Watcher) ValidSegment(segmentNum int) (bool, error) {
+	segments, err := w.segments(w.walDir)
+	if err != nil {
+		return false, err
+	}
+	for _, s := range segments {
+		if s == segmentNum {
+			return true, nil
+		}
+	}
```

Comment:
Since the segments are sorted you may be able to do an early return here if `s > segmentNum`.